### ISSUE TITLE
MonteCarlo statistics are wrong

### ISSUE THREAD

**Issue by yannickspill** [salilab/imp#765]
MonteCarlo has a couple of functions to get statistics
`get_number_of_forward_steps`,  `get_number_of_upward_steps`, `get_number_of_proposed_steps` and `get_number_of_accepted_steps`.

What would be expected is that if one calls `mc.optimize(N)`, then
- `get_number_of_proposed_steps` returns `N`
- `get_number_of_accepted_steps` returns the number of times moves succeeded
- `get_number_of_upward_steps` returns the number of steps to higher scores.

I'm clueless about `get_number_of_forward_steps`. Should that be `get_number_of_downward_steps` instead, i.e. the number of steps to lower scores? In that case, in `MonteCarlo.cpp`, `++stat_forward_steps_taken_` should be moved from line 58 to line 38. Otherwise it would be synonymous to `get_number_of_accepted_steps`, which seems to contradict its implementation. For now, `get_number_of_accepted_steps` and `get_number_of_proposed_steps` return bogus values, because `stat_forward_steps_taken_+stat_upward_steps_taken_` might be larger than `N` in some cases.

In short, I propose to
- move  `++stat_forward_steps_taken_` from line 58 to line 38 in `MonteCarlo.cpp`
- rename `get_number_of_forward_steps` to `get_number_of_downward_steps` to be coherent with `get_number_of_upward_steps`
